### PR TITLE
Adds pushing images to dockerhub as well

### PR DESCRIPTION
**Pull request by s1113950**
Related to https://github.com/kubeflow/kubeflow/issues/211

This is my first contribution to kubeflow, and as such I didn't want to mess with how the Makefiles are laid out (they aren't standardized). Perhaps that's a future task?

I ran into this when trying to build `k8s-model-server/http-proxy` locally:

```
Step 5/8 : COPY requirements.txt /usr/src/app/
COPY failed: stat /var/lib/docker/tmp/docker-builder102266424/requirements.txt: no such file or directory
```
I didn't look into it in detail though. Given that I only made makefile targets that were the same as the previous commands, I don't feel like this is related but just want to make sure.

### CODE REVIEW

**Comment by s1113950** [comment]
testing cla check

File: components/k8s-model-server/http-proxy/Makefile [hunk 1]
```diff
@@ -1,16 +1,45 @@
+# Copyright 2017 Google Inc.
```
Comment (carmark) [inline components/k8s-model-server/http-proxy/Makefile]: 2018?

File: components/tensorflow-notebook-image/Makefile [hunk 2]
```diff
@@ -11,18 +11,34 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 
-all: container
-
 TAG?=$(shell git rev-parse HEAD)
-REGISTRY?=gcr.io/kubeflow
+GCR_REGISTRY?=gcr.io/kubeflow
+DOCKERHUB_REGISTRY?=hub.docker.com/r/kubeflow
```
Comment (carmark) [inline components/tensorflow-notebook-image/Makefile]: How about using the `PROJECT_ID` macros?

File: testing/Makefile [hunk 3]
```diff
@@ -12,20 +12,38 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-IMG = gcr.io/mlkube-testing/kubeflow-testing
-TAG := $(shell date +v%Y%m%d)-$(shell git describe --tags --always --dirty)-$(shell git diff | sha256sum | cut -c -6)
-DIR := ${CURDIR}
 
-all: build
+GCR_IMG = gcr.io/mlkube-testing/kubeflow-testing
+DOCKERHUB_IMG = hub.docker.com/r/kubeflow-testing
```
Comment (carmark) [inline testing/Makefile]: same as above.

File: tf-controller-examples/tf-cnn/Makefile [hunk 4]
```diff
@@ -12,8 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-REGISTRY := gcr.io/kubeflow
-TAG := $(shell date +v%Y%m%d)-$(shell git describe --tags --always --dirty)-$(shell git diff | sha256sum | cut -c -6)
+GCR_REGISTRY := gcr.io/kubeflow
+DOCKERHUB_REGISTRY := hub.docker.com/r/kubeflow
```
Comment (carmark) [inline tf-controller-examples/tf-cnn/Makefile]: same as above.

**Comment by jlewi** [comment]
Is there a way to mirror one docker registry (gcr) to another (DockerHub) so we don't need to push to two locations everywhere?

What are the relevant permissions ownership of the DockerHub repo this is pushing to?

File: components/k8s-model-server/http-proxy/Makefile (hunk 1 above)
Comment (s1113950) [reply components/k8s-model-server/http-proxy/Makefile]: ahh, good catch. I noticed this file didn't have a header so I copied it over from another file without changing the year

File: components/tensorflow-notebook-image/Makefile (hunk 2 above)
Comment (s1113950) [reply components/tensorflow-notebook-image/Makefile]: I can do that. I noticed that each Makefile already was using different variables; I can standardize it to all use PROJECT_ID though

**Comment by s1113950** [comment]
With the exception of `testing`, which I noticed was going to `mlkube-testing/kubeflow-testing`, all of them go to `hub.docker.com/r/kubeflow`. 
I don't have permission to push to this repo: https://hub.docker.com/r/kubeflow/ 
I saw on the original ticket @jlewi you had a comment here: `See #197 for getting ownership of kubeflow in Dockerhub.`; in testing this I pushed to our own Dockerhub instead by overriding the `DOCKERHUB_PROJECT` var during each make call. Presumably @carmark can change permissions to whatever is required for this to work? see https://github.com/kubeflow/kubeflow/issues/197 for more info

https://cloud.google.com/container-registry/docs/using-dockerhub-mirroring this is what I've found so far regarding mirroring docker registries. It looks like we'd follow the section under `Fetch images from Container Registry's mirror`. I'm a bit confused by it though. It looks like it's all related to a local docker daemon? So presumably we might not have to do anything on our end at all other than maybe a script that sets up that config for a user in their local docker daemon?

**Comment by s1113950** [comment]
@carmark I completed the revisions requested

**Comment by jlewi** [comment]
I updated the PR description because this is only a partial fix since we will actually have to rebuild the images and push them. There's also work to do setting up the DockerHub registry and we should probably move the GCR registry to a project that can be owned by the community.

**Comment by carmark** [comment]
@s1113950 The registry mirror is used in docker daemon, so we can not use it to sync the image between docker hub and gcr.

**Pull request merged**